stock_barcodes/wizard/stock_barcodes_read.py

```diff
@@ -0,0 +1,193 @@
+        string='Lot scanned',
+    )
+    location_id = fields.Many2one(
+        comodel_name='stock.location',
+        string='Location',
+    )
+    packaging_id = fields.Many2one(
+        comodel_name='product.packaging',
+        string='Packaging',
+    )
+    packaging_qty = fields.Float(
+        string='Package Qty',
+        digits=dp.get_precision('Product Unit of Measure'),
+    )
+    product_qty = fields.Float(
+        string='Product Qty',
+        digits=dp.get_precision('Product Unit of Measure'),
+    )
+    manual_entry = fields.Boolean(
+        string='Manual entry data',
+    )
+    # Computed field for display all scan logs from res_model and res_id when
+    # change product_id
+    scan_log_ids = fields.Many2many(
+        comodel_name='stock.barcodes.read.log',
+        compute='_compute_scan_log_ids',
+    )
+    message_type = fields.Selection([
+        ('info', 'Barcode read with additional info'),
+        ('not_found', 'No barcode found'),
+        ('more_match', 'More than one matches found'),
+        ('success', 'Barcode read correctly'),
+    ], readonly=True)
+    message = fields.Char(readonly=True)
+
+    @api.onchange('location_id')
+    def onchange_location_id(self):
+        self.packaging_id = False
+        self.product_id = False
+
+    @api.onchange('packaging_qty')
+    def onchange_packaging_qty(self):
+        if self.packaging_id:
+            self.product_qty = self.packaging_qty * self.packaging_id.qty
+
+    def _set_messagge_info(self, type, messagge):
+        """
+        Set message type and message description.
+        For manual entry mode barcode is not set so is not displayed
+        """
+        self.message_type = type
+        if self.barcode:
+            self.message = '%s (%s)' % (messagge, self.barcode)
+        else:
+            self.message = '%s' % messagge
+
+    def process_barcode(self, barcode):
+        self._set_messagge_info('success', _('Barcode read correctly'))
+        domain = self._barcode_domain(barcode)
+        product = self.env['product.product'].search(domain)
+        if product:
+            if len(product) > 1:
+                self._set_messagge_info(
+                    'more_match', _('More than one product found'))
+                return
+            self.action_product_scaned_post(product)
+            self.action_done()
+            return
+        packaging = self.env['product.packaging'].search(domain)
+        if packaging:
+            if len(packaging) > 1:
+                self._set_messagge_info(
+                    'more_match', _('More than one package found'))
+                return
+            self.action_packaging_scaned_post(packaging)
+            self.action_done()
+            return
+        lot = self.env['stock.production.lot'].search([
+            ('name', '=', barcode),
+            ('product_id', '=', self.product_id.id),
+        ])
+        if lot:
+            self.lot_id = lot
+            return
```

Comment:
I think is good idea set qty after scan lot.
Example:
I have product traking by serial number.
I do inventory adjastment (steps start after open wiz.stock.barcodes.read from view):
1. scan product
2. scan lot
3. scan product
count_steps = (count_lots * 2) + 1
If add function like:
> def action_lot_scaned_post(self):
>> self.product_qty = 0.0 if self.manual_entry else 1.0

and call it with action_done()

In the same case we have steps:
1. scan product
2. scan lot
count_steps = count_lots + 1
@pedrobaeza What do you think about it?

Reply:
Is other point of view... I will study the case for product with tracking 'serial'.

Reply:
Ok.. 
I have implemented here https://github.com/OCA/stock-logistics-barcode/pull/195/commits/8a852ca3da3e5a277aaeb2b2d6f126270ddfd1f6
Thanks!!